src/attpc_engine/detector/simulator.py

```diff
@@ -254,7 +293,9 @@ def generate_electrons(
         )
         return electrons
 
-    def generate_point_cloud(self, config: Config, rng: Generator) -> np.ndarray:
+    def generate_point_cloud(
```

Comment:
Not clear what is happening here. We give it a dictionary to fill and then you return the same dictionary? Should just accept the dict, fill it.

Reply:
Also need type hint for points. Should be 
```python
points: numba.typed.Dict[int, int]
```